carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_guidance

```diff
@@ -32,14 +32,18 @@ def check_plugin_status(plugin_msg):
         plugins_activated = True
 
 def initialize():
+    rospy.init_node("carma_carla_guidance")
     service = rospy.ServiceProxy('/set_active_guidance')
-    route_sub = rospy.Subscriber("/route_event",RouteEvent(), check_route_status)
-    plugin_sub = rospy.Subscriber("/plugin_discovery", Plugin(), check_plugin_status)
-    if route_selected == True:
-        if plugins_activated == True:
-            service.call()
-    else:
-        rospy.spin()
+    #While-loop monitors the route and plugin status and sets guidance to active once both are confirmed true
+    while rospy.is_shutdown() != True:
+        route_sub = rospy.Subscriber("/route_event",RouteEvent(), check_route_status)
+        plugin_sub = rospy.Subscriber("/plugin_discovery", Plugin(), check_plugin_status)
+        if route_selected == True:
```

Comment:
It could be simplified by adding `and` here. For example, `if route_selected and plugins_activated`